crates/bevy_reflect/src/lib.rs

```diff
@@ -1993,7 +2008,7 @@ bevy_reflect::tests::Test {
         }
 
         let foo = Foo(123);
-        let foo: &dyn Reflect = &foo;
+        let foo: &dyn PartialReflect = &foo;
```

Comment:
Any particular reason for changing things like this? I know it's just a variable in a test haha, but I'm curious if this was a requirement as that could affect user migration.